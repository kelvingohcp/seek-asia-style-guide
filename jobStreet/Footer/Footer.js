import React from 'react';
import styles from './footer.less';
import PropTypes from 'prop-types';
import FooterLinks from './components/FooterLinks/FooterLinks';
import { getLocalization } from '../localization';
import _get from 'lodash/get';

const Footer = ({ language, country }) => {
  const year = new Date().getFullYear();
  const messages = getLocalization({ country, language });
  const footerMessages = _get(messages, 'footer');

  if (footerMessages) {
    return (
      <footer className={styles.container}>
        <FooterLinks footerMessages={footerMessages} />
        <p className={styles.copyright}>
          {_get(footerMessages, 'copyright').replace('{year}', year)}
        </p>
      </footer>
    );
  }

  return null;
};

Footer.propTypes = {
  language: PropTypes.string.isRequired,
  country: PropTypes.string.isRequired
};

export default Footer;
