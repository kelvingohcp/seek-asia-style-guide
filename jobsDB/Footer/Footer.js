import React from 'react';
import styles from './footer.less';
import PropTypes from 'prop-types';

import FooterLinks from './components/FooterLinks/FooterLinks';
import FooterHK from './FooterHK/FooterHK';
import localization from './localization';

const Footer = ({ language, country, ...restProps }) => {
  const year = new Date().getFullYear();
  const messages = localization[`${language}-${country}`];

  if (country === 'hk') {
    return (<FooterHK language={language} country={country} {...restProps} />);
  }
  return (
    <footer className={styles.container}>
      <div className={styles.wrapper}>
        <div className={styles.pullRight}>
          <FooterLinks messages={messages} />
        </div>
        <div className={styles.pullLeft}>
          <p className={styles.copyright}>
            {messages['footer.copyright'].replace('{year}', year)}
            <span className={styles.fullversion}>
              {messages['footer.rightReserved']}
            </span>
          </p>
        </div>
      </div>
    </footer>
  );
};

Footer.propTypes = {
  language: PropTypes.string.isRequired,
  country: PropTypes.string.isRequired
};

export default Footer;
