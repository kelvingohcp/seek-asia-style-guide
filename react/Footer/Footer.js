import React from 'react';
import PropTypes from 'prop-types';
import GlobalFooter from '../private/GlobalFooter/GlobalFooter';
import { getLocalization } from '../../jobsDB/localization';
import { getLocalList } from '../../jobsDB/helper';
import _get from 'lodash/get';

const Footer = ({
  language,
  country,
  isExpandedVersion,
  showHeaderActionTrayOffset,
  showCountryLanguage,
  domainUrl,
  ...restProps }) => {
  const messages = getLocalization({ country, language });
  const footerMessages = _get(messages, 'footer');

  if (footerMessages) {
    return (
      <GlobalFooter
        country={country}
        isExpandedVersion={isExpandedVersion}
        showHeaderActionTrayOffset={showHeaderActionTrayOffset}
        showCountryLanguage={showCountryLanguage}
        footerMessages={footerMessages}
        locales={getLocalList({ messages, domainUrl, country, language })}
        {...restProps}
      />
    );
  }
  return null;
};

Footer.defaultProps = {
  country: 'hk',
  language: 'en',
  domainUrl: 'jobsdb.com',
  hasCompanyProfile: true,
  showHeaderActionTrayOffset: true,
  isExpandedVersion: false,
  showCountryLanguage: true
};

Footer.propTypes = {
  country: PropTypes.string.isRequired,
  language: PropTypes.string.isRequired,
  domainUrl: PropTypes.string.isRequired,
  hasCompanyProfile: PropTypes.bool,
  showHeaderActionTrayOffset: PropTypes.bool,
  isExpandedVersion: PropTypes.bool,
  showCountryLanguage: PropTypes.bool
};

export default Footer;
