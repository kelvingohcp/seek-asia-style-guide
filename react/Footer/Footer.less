@import (reference) "~seek-asia-style-guide/theme";
@import (reference) "~seek-asia-style-guide/react/Header/components/ActionTray/ActionTray.less";
@wrapper-up-padding-top: @grid-base * 8;
@wrapper-up-padding-bottom: @grid-base * 3;

.container {
  cursor: default;
  margin: 0 auto;
  background-color: @saGrey6;
  width: 100%;
  font-weight: @sa-semi-bold;
}

.headerActionTrayOffset {
  @media @mobile {
    margin-bottom: @actionTrayHeight;
  }
}

.link {
  text-decoration: none;
}

.upperWrapper {
  padding-top: @wrapper-up-padding-top;
  padding-bottom: @wrapper-up-padding-bottom;

  -webkit-transition: all 0.3s ease-in-out;
  -moz-transition: all 0.3s ease-in-out;
  -ms-transition: all 0.3s ease-in-out;
  -o-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;

  &.hidden {
    max-height: 0;
    padding-top: 0;
    padding-bottom: 0;
    overflow: hidden;
    opacity: 0;
  }
}

.lowerWrapper {
  display: flex;
  justify-content: space-between;
  align-items: stretch;
}

.lowerWrapperLinks {
  display: flex;
  flex-direction: column;
  line-height: @grid-base * 4;
  padding-left: @grid-base * 2;
  width: 100%;
  @media @tablet {
    flex-direction: row;
    line-height: normal;
    padding-bottom: @grid-base;
    width: 100%;
  }
  @media @desktop {
    padding-bottom: @grid-base;
    width: auto;
  }
}

.fullWidth {
  width: 100%;
}

.collapsed {
  padding-left: @grid-base * 4;
  @media @desktop {
    padding-left: @grid-base * 6;
  }
}

.chevronSvg {
  height: @grid-base * 3;
  width: @grid-base * 3;
}

.chevronIcon {
  cursor: pointer;
  padding-right: @grid-base * 4;
  padding-left: @grid-base * 8;
  display: flex;
  align-items: center;
  color: @saBlue2;
  @media @desktop {
    padding-right: @grid-base * 6;
    padding-left: @grid-base * 12;
  }
}

.endOfLine {
  display: flex;
  width: 100%;
  @media @tablet {
    width: 100%;
  }
  @media @desktop {
    width: auto;
  }
}

.endOfLineWithoutCountry {
  display: flex;
  width: 100%;
}

.localeIcon {
  margin: @grid-base @grid-base 0 0;
}

.currentLocale {
  display: none;
  @media @desktop {
    display: flex;
  }
}
