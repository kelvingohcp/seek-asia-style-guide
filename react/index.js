export { default as StyleGuideProvider } from './StyleGuideProvider/StyleGuideProvider';
export { default as SeekApp } from './StyleGuideProvider/StyleGuideProvider'; // Alias for backwards compatibility

// Components
export { default as Alert } from './Alert/Alert';
export { default as Button } from './Button/Button';
export { default as FilterButton } from './FilterButton/FilterButton';
export { default as ButtonGroup } from './ButtonGroup/ButtonGroup';
export { default as ListItem } from './ListItem/ListItem';
export { default as Loader } from './Loader/Loader';
export { default as Logo } from './Logo/Logo';
export { default as LogoRainbow } from './LogoRainbow/LogoRainbow';
export { default as Rating } from './Rating/Rating';
export { default as NavigationBar } from './NavigationBar/NavigationBar';
export { default as JobCard, trackLinkType } from './JobCard/JobCard';
export { default as JobCardLoader } from './JobCardLoader/JobCardLoader';
export { default as CardLoader } from './CardLoader/CardLoader';
export { default as CardLoadingBar } from './CardLoadingBar/CardLoadingBar';
export { default as Header } from './Header/Header';
export { default as ShowMore } from './ShowMore/ShowMore';
export { default as TitledCard } from './TitledCard/TitledCard';
export { default as Icon } from './Icon/Icon';
export { default as DropdownList } from './DropdownList/DropdownList';
export { default as Tag } from './Tag/Tag';
export { default as Badge } from './Badge/Badge';

// Typography
export { default as Critical } from './Critical/Critical';
export { default as Paragraph } from './Paragraph/Paragraph';
export { default as Positive } from './Positive/Positive';
export { default as Secondary } from './Secondary/Secondary';
export { default as Strong } from './Strong/Strong';
export { default as Text } from './Text/Text';
export { default as TextLink } from './TextLink/TextLink';

// Layout
export { default as AsidedLayout } from './AsidedLayout/AsidedLayout';
export { default as Card } from './Card/Card';
export { default as CardGroup } from './CardGroup/CardGroup';
export { default as Columns } from './Columns/Columns';
export { default as Hidden } from './Hidden/Hidden';
export { default as PageBlock } from './PageBlock/PageBlock';
export { default as Section } from './Section/Section';

// Icons
export { default as CriticalIcon } from './CriticalIcon/CriticalIcon';
export { default as CrossIcon } from './CrossIcon/CrossIcon';
export { default as ErrorIcon } from './ErrorIcon/ErrorIcon';
export { default as FieldOfStudyIcon } from './FieldOfStudyIcon/FieldOfStudyIcon';
export { default as FlagHKIcon } from './FlagHKIcon/FlagHKIcon';
export { default as FlagIDIcon } from './FlagIDIcon/FlagIDIcon';
export { default as FlagMYIcon } from './FlagMYIcon/FlagMYIcon';
export { default as FlagPHIcon } from './FlagPHIcon/FlagPHIcon';
export { default as FlagSGIcon } from './FlagSGIcon/FlagSGIcon';
export { default as FlagTHIcon } from './FlagTHIcon/FlagTHIcon';
export { default as FlagVNIcon } from './FlagVNIcon/FlagVNIcon';
export { default as HelpIcon } from './HelpIcon/HelpIcon';
export { default as InfoIcon } from './InfoIcon/InfoIcon';
export { default as JobFunctionIcon } from './JobFunctionIcon/JobFunctionIcon';
export { default as JobInvitationIcon } from './JobInvitationIcon/JobInvitationIcon';
export { default as MoreIcon } from './MoreIcon/MoreIcon';
export { default as ProfileIcon } from './ProfileIcon/ProfileIcon';
export { default as StarIcon } from './StarIcon/StarIcon';
export { default as TickCircleIcon } from './TickCircleIcon/TickCircleIcon';

// Fields
export { default as Autosuggest } from './Autosuggest/Autosuggest';
export { default as Checkbox } from './Checkbox/Checkbox';
export { default as ClearField } from './ClearField/ClearField';
export { default as Dropdown } from './Dropdown/Dropdown';
export { default as EmailField } from './EmailField/EmailField';
export { default as MonthPicker } from './MonthPicker/MonthPicker';
export { default as SlideToggle } from './SlideToggle/SlideToggle';
export { default as Textarea } from './Textarea/Textarea';
export { default as TextField } from './TextField/TextField';

// Accessibility
export { default as ScreenReaderOnly } from './ScreenReaderOnly/ScreenReaderOnly';
export { default as ScreenReaderSkipLink } from './ScreenReaderSkipLink/ScreenReaderSkipLink';
export { default as ScreenReaderSkipTarget } from './ScreenReaderSkipTarget/ScreenReaderSkipTarget';

// Constants
export { default as Constants } from './Constants/Constants';
