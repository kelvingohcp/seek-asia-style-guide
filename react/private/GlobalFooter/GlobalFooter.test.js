import React from 'react';
import { shallow } from 'enzyme';
import footerMessages from './mock/footerMessage';
import Footer from './GlobalFooter';
import { FlagHKIcon } from 'seek-asia-style-guide/react';

describe('Footer component', () => {
  const defaultProps = {
    country: 'hk',
    language: 'en',
    domainUrl: 'jobsdb.com',
    footerMessages,
    locales: [
      {
        title: 'Hong Kong (English)',
        ItemIcon: FlagHKIcon,
        url: '',
        language: 'en',
        country: 'hk',
        languageDisplay: 'English'
      }
    ]
  };

  it('should render with default props', () => {
    const wrapper = shallow(<Footer {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with custom link renderer', () => {
    const wrapper = shallow(<Footer linkRenderer={() => {}} {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with custom href link', () => {
    const wrapper = shallow(
      <Footer
        hrefLink={({ link }) => {
          return link;
        }} {...defaultProps}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
