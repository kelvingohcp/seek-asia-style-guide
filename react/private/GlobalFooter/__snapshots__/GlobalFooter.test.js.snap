// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render with custom href link 1`] = `
<footer
  className="container"
>
  <PageBlock>
    <div
      className="upperWrapper hidden"
    >
      <UpperFooter
        footerMessages={
          Object {
            "about": Object {
              "about": Object {
                "title": "About jobsDB",
                "url": "/pages/aboutus",
              },
              "contactUs": Object {
                "mobileUrl": "https://m.jobsdb.com/en-hk/legal.do?p=contact_us",
                "title": "Contact us",
                "url": "/pages/regional-coverage",
              },
              "title": "About JobsDB",
            },
            "cfsLink": "/hk/en",
            "cmsLink": "/en-hk",
            "connect": Object {
              "appStore": Object {
                "icon": "apple",
                "title": "jobsDB @ App Store",
                "url": "https://itunes.apple.com/hk/app/jobsdb-job-search/id414607432",
              },
              "playStore": Object {
                "icon": "android",
                "title": "jobsDB @ Google Play",
                "url": "https://play.google.com/store/apps/details?id=com.jobsdb",
              },
              "title": "Connect",
            },
            "copyright": "Copyright © 1998-{year}, jobsDB.",
            "legal": Object {
              "title": "Legal terms",
              "url": "/legal.do",
            },
            "partners": Object {
              "bdjobs": Object {
                "title": "Bdjobs (Bandladesh)",
                "url": "https://www.bdjobs.com/",
              },
              "brighterMonday": Object {
                "title": "Brighter Monday (East Africa)",
                "url": "https://www.brightermonday.com/",
              },
              "catho": Object {
                "title": "Catho (Brazil)",
                "url": "https://www.catho.com.br/",
              },
              "title": "International partners",
              "zhaopin": Object {
                "title": "Zhaopin (China)",
                "url": "https://www.zhaopin.com/",
              },
            },
            "privacy": Object {
              "title": "Privacy Statement",
              "url": "/pages/terms/privacy-policy",
            },
            "rightReserved": "All Rights Reserved.",
            "siteMap": Object {
              "title": "Site Map",
              "url": "/sitemap",
            },
            "terms": Object {
              "title": "Terms & Conditions",
              "url": "/pages/terms/terms-conditions",
            },
          }
        }
        linkRenderer={[Function]}
      />
      <Hidden
        aboveMobile={true}
        className=""
        component="span"
        desktop={false}
        mobile={false}
        print={false}
        screen={false}
      >
        <FooterLinks
          footerMessages={
            Object {
              "about": Object {
                "about": Object {
                  "title": "About jobsDB",
                  "url": "/pages/aboutus",
                },
                "contactUs": Object {
                  "mobileUrl": "https://m.jobsdb.com/en-hk/legal.do?p=contact_us",
                  "title": "Contact us",
                  "url": "/pages/regional-coverage",
                },
                "title": "About JobsDB",
              },
              "cfsLink": "/hk/en",
              "cmsLink": "/en-hk",
              "connect": Object {
                "appStore": Object {
                  "icon": "apple",
                  "title": "jobsDB @ App Store",
                  "url": "https://itunes.apple.com/hk/app/jobsdb-job-search/id414607432",
                },
                "playStore": Object {
                  "icon": "android",
                  "title": "jobsDB @ Google Play",
                  "url": "https://play.google.com/store/apps/details?id=com.jobsdb",
                },
                "title": "Connect",
              },
              "copyright": "Copyright © 1998-{year}, jobsDB.",
              "legal": Object {
                "title": "Legal terms",
                "url": "/legal.do",
              },
              "partners": Object {
                "bdjobs": Object {
                  "title": "Bdjobs (Bandladesh)",
                  "url": "https://www.bdjobs.com/",
                },
                "brighterMonday": Object {
                  "title": "Brighter Monday (East Africa)",
                  "url": "https://www.brightermonday.com/",
                },
                "catho": Object {
                  "title": "Catho (Brazil)",
                  "url": "https://www.catho.com.br/",
                },
                "title": "International partners",
                "zhaopin": Object {
                  "title": "Zhaopin (China)",
                  "url": "https://www.zhaopin.com/",
                },
              },
              "privacy": Object {
                "title": "Privacy Statement",
                "url": "/pages/terms/privacy-policy",
              },
              "rightReserved": "All Rights Reserved.",
              "siteMap": Object {
                "title": "Site Map",
                "url": "/sitemap",
              },
              "terms": Object {
                "title": "Terms & Conditions",
                "url": "/pages/terms/terms-conditions",
              },
            }
          }
          linkRenderer={[Function]}
        />
      </Hidden>
    </div>
    <div
      className="lowerWrapper"
    >
      <div
        className="meta"
      >
        <Hidden
          aboveMobile={false}
          className=""
          component="span"
          desktop={false}
          mobile={true}
          print={false}
          screen={false}
        >
          <FooterLinks
            className="bottomMeta"
            footerMessages={
              Object {
                "about": Object {
                  "about": Object {
                    "title": "About jobsDB",
                    "url": "/pages/aboutus",
                  },
                  "contactUs": Object {
                    "mobileUrl": "https://m.jobsdb.com/en-hk/legal.do?p=contact_us",
                    "title": "Contact us",
                    "url": "/pages/regional-coverage",
                  },
                  "title": "About JobsDB",
                },
                "cfsLink": "/hk/en",
                "cmsLink": "/en-hk",
                "connect": Object {
                  "appStore": Object {
                    "icon": "apple",
                    "title": "jobsDB @ App Store",
                    "url": "https://itunes.apple.com/hk/app/jobsdb-job-search/id414607432",
                  },
                  "playStore": Object {
                    "icon": "android",
                    "title": "jobsDB @ Google Play",
                    "url": "https://play.google.com/store/apps/details?id=com.jobsdb",
                  },
                  "title": "Connect",
                },
                "copyright": "Copyright © 1998-{year}, jobsDB.",
                "legal": Object {
                  "title": "Legal terms",
                  "url": "/legal.do",
                },
                "partners": Object {
                  "bdjobs": Object {
                    "title": "Bdjobs (Bandladesh)",
                    "url": "https://www.bdjobs.com/",
                  },
                  "brighterMonday": Object {
                    "title": "Brighter Monday (East Africa)",
                    "url": "https://www.brightermonday.com/",
                  },
                  "catho": Object {
                    "title": "Catho (Brazil)",
                    "url": "https://www.catho.com.br/",
                  },
                  "title": "International partners",
                  "zhaopin": Object {
                    "title": "Zhaopin (China)",
                    "url": "https://www.zhaopin.com/",
                  },
                },
                "privacy": Object {
                  "title": "Privacy Statement",
                  "url": "/pages/terms/privacy-policy",
                },
                "rightReserved": "All Rights Reserved.",
                "siteMap": Object {
                  "title": "Site Map",
                  "url": "/sitemap",
                },
                "terms": Object {
                  "title": "Terms & Conditions",
                  "url": "/pages/terms/terms-conditions",
                },
              }
            }
            linkRenderer={[Function]}
          />
        </Hidden>
        <Text
          baseline={true}
          className="copyright"
          secondary={true}
          semiStrong={true}
          whispering={true}
        >
          Copyright © 1998-2020, jobsDB.
        </Text>
      </div>
      <div
        className="chevronIcon"
        onClick={[Function]}
      >
        <Icon
          animation=""
          id="pageBottom"
          lineHeight=""
          rotation="-180deg"
          size=""
          smoothRotate={true}
          type="chevron"
        />
      </div>
    </div>
  </PageBlock>
</footer>
`;

exports[`Footer component should render with custom link renderer 1`] = `
<footer
  className="container"
>
  <PageBlock>
    <div
      className="upperWrapper hidden"
    >
      <UpperFooter
        footerMessages={
          Object {
            "about": Object {
              "about": Object {
                "title": "About jobsDB",
                "url": "/pages/aboutus",
              },
              "contactUs": Object {
                "mobileUrl": "https://m.jobsdb.com/en-hk/legal.do?p=contact_us",
                "title": "Contact us",
                "url": "/pages/regional-coverage",
              },
              "title": "About JobsDB",
            },
            "cfsLink": "/hk/en",
            "cmsLink": "/en-hk",
            "connect": Object {
              "appStore": Object {
                "icon": "apple",
                "title": "jobsDB @ App Store",
                "url": "https://itunes.apple.com/hk/app/jobsdb-job-search/id414607432",
              },
              "playStore": Object {
                "icon": "android",
                "title": "jobsDB @ Google Play",
                "url": "https://play.google.com/store/apps/details?id=com.jobsdb",
              },
              "title": "Connect",
            },
            "copyright": "Copyright © 1998-{year}, jobsDB.",
            "legal": Object {
              "title": "Legal terms",
              "url": "/legal.do",
            },
            "partners": Object {
              "bdjobs": Object {
                "title": "Bdjobs (Bandladesh)",
                "url": "https://www.bdjobs.com/",
              },
              "brighterMonday": Object {
                "title": "Brighter Monday (East Africa)",
                "url": "https://www.brightermonday.com/",
              },
              "catho": Object {
                "title": "Catho (Brazil)",
                "url": "https://www.catho.com.br/",
              },
              "title": "International partners",
              "zhaopin": Object {
                "title": "Zhaopin (China)",
                "url": "https://www.zhaopin.com/",
              },
            },
            "privacy": Object {
              "title": "Privacy Statement",
              "url": "/pages/terms/privacy-policy",
            },
            "rightReserved": "All Rights Reserved.",
            "siteMap": Object {
              "title": "Site Map",
              "url": "/sitemap",
            },
            "terms": Object {
              "title": "Terms & Conditions",
              "url": "/pages/terms/terms-conditions",
            },
          }
        }
        linkRenderer={[Function]}
      />
      <Hidden
        aboveMobile={true}
        className=""
        component="span"
        desktop={false}
        mobile={false}
        print={false}
        screen={false}
      >
        <FooterLinks
          footerMessages={
            Object {
              "about": Object {
                "about": Object {
                  "title": "About jobsDB",
                  "url": "/pages/aboutus",
                },
                "contactUs": Object {
                  "mobileUrl": "https://m.jobsdb.com/en-hk/legal.do?p=contact_us",
                  "title": "Contact us",
                  "url": "/pages/regional-coverage",
                },
                "title": "About JobsDB",
              },
              "cfsLink": "/hk/en",
              "cmsLink": "/en-hk",
              "connect": Object {
                "appStore": Object {
                  "icon": "apple",
                  "title": "jobsDB @ App Store",
                  "url": "https://itunes.apple.com/hk/app/jobsdb-job-search/id414607432",
                },
                "playStore": Object {
                  "icon": "android",
                  "title": "jobsDB @ Google Play",
                  "url": "https://play.google.com/store/apps/details?id=com.jobsdb",
                },
                "title": "Connect",
              },
              "copyright": "Copyright © 1998-{year}, jobsDB.",
              "legal": Object {
                "title": "Legal terms",
                "url": "/legal.do",
              },
              "partners": Object {
                "bdjobs": Object {
                  "title": "Bdjobs (Bandladesh)",
                  "url": "https://www.bdjobs.com/",
                },
                "brighterMonday": Object {
                  "title": "Brighter Monday (East Africa)",
                  "url": "https://www.brightermonday.com/",
                },
                "catho": Object {
                  "title": "Catho (Brazil)",
                  "url": "https://www.catho.com.br/",
                },
                "title": "International partners",
                "zhaopin": Object {
                  "title": "Zhaopin (China)",
                  "url": "https://www.zhaopin.com/",
                },
              },
              "privacy": Object {
                "title": "Privacy Statement",
                "url": "/pages/terms/privacy-policy",
              },
              "rightReserved": "All Rights Reserved.",
              "siteMap": Object {
                "title": "Site Map",
                "url": "/sitemap",
              },
              "terms": Object {
                "title": "Terms & Conditions",
                "url": "/pages/terms/terms-conditions",
              },
            }
          }
          linkRenderer={[Function]}
        />
      </Hidden>
    </div>
    <div
      className="lowerWrapper"
    >
      <div
        className="meta"
      >
        <Hidden
          aboveMobile={false}
          className=""
          component="span"
          desktop={false}
          mobile={true}
          print={false}
          screen={false}
        >
          <FooterLinks
            className="bottomMeta"
            footerMessages={
              Object {
                "about": Object {
                  "about": Object {
                    "title": "About jobsDB",
                    "url": "/pages/aboutus",
                  },
                  "contactUs": Object {
                    "mobileUrl": "https://m.jobsdb.com/en-hk/legal.do?p=contact_us",
                    "title": "Contact us",
                    "url": "/pages/regional-coverage",
                  },
                  "title": "About JobsDB",
                },
                "cfsLink": "/hk/en",
                "cmsLink": "/en-hk",
                "connect": Object {
                  "appStore": Object {
                    "icon": "apple",
                    "title": "jobsDB @ App Store",
                    "url": "https://itunes.apple.com/hk/app/jobsdb-job-search/id414607432",
                  },
                  "playStore": Object {
                    "icon": "android",
                    "title": "jobsDB @ Google Play",
                    "url": "https://play.google.com/store/apps/details?id=com.jobsdb",
                  },
                  "title": "Connect",
                },
                "copyright": "Copyright © 1998-{year}, jobsDB.",
                "legal": Object {
                  "title": "Legal terms",
                  "url": "/legal.do",
                },
                "partners": Object {
                  "bdjobs": Object {
                    "title": "Bdjobs (Bandladesh)",
                    "url": "https://www.bdjobs.com/",
                  },
                  "brighterMonday": Object {
                    "title": "Brighter Monday (East Africa)",
                    "url": "https://www.brightermonday.com/",
                  },
                  "catho": Object {
                    "title": "Catho (Brazil)",
                    "url": "https://www.catho.com.br/",
                  },
                  "title": "International partners",
                  "zhaopin": Object {
                    "title": "Zhaopin (China)",
                    "url": "https://www.zhaopin.com/",
                  },
                },
                "privacy": Object {
                  "title": "Privacy Statement",
                  "url": "/pages/terms/privacy-policy",
                },
                "rightReserved": "All Rights Reserved.",
                "siteMap": Object {
                  "title": "Site Map",
                  "url": "/sitemap",
                },
                "terms": Object {
                  "title": "Terms & Conditions",
                  "url": "/pages/terms/terms-conditions",
                },
              }
            }
            linkRenderer={[Function]}
          />
        </Hidden>
        <Text
          baseline={true}
          className="copyright"
          secondary={true}
          semiStrong={true}
          whispering={true}
        >
          Copyright © 1998-2020, jobsDB.
        </Text>
      </div>
      <div
        className="chevronIcon"
        onClick={[Function]}
      >
        <Icon
          animation=""
          id="pageBottom"
          lineHeight=""
          rotation="-180deg"
          size=""
          smoothRotate={true}
          type="chevron"
        />
      </div>
    </div>
  </PageBlock>
</footer>
`;

exports[`Footer component should render with default props 1`] = `
<footer
  className="container"
>
  <PageBlock>
    <div
      className="upperWrapper hidden"
    >
      <UpperFooter
        footerMessages={
          Object {
            "about": Object {
              "about": Object {
                "title": "About jobsDB",
                "url": "/pages/aboutus",
              },
              "contactUs": Object {
                "mobileUrl": "https://m.jobsdb.com/en-hk/legal.do?p=contact_us",
                "title": "Contact us",
                "url": "/pages/regional-coverage",
              },
              "title": "About JobsDB",
            },
            "cfsLink": "/hk/en",
            "cmsLink": "/en-hk",
            "connect": Object {
              "appStore": Object {
                "icon": "apple",
                "title": "jobsDB @ App Store",
                "url": "https://itunes.apple.com/hk/app/jobsdb-job-search/id414607432",
              },
              "playStore": Object {
                "icon": "android",
                "title": "jobsDB @ Google Play",
                "url": "https://play.google.com/store/apps/details?id=com.jobsdb",
              },
              "title": "Connect",
            },
            "copyright": "Copyright © 1998-{year}, jobsDB.",
            "legal": Object {
              "title": "Legal terms",
              "url": "/legal.do",
            },
            "partners": Object {
              "bdjobs": Object {
                "title": "Bdjobs (Bandladesh)",
                "url": "https://www.bdjobs.com/",
              },
              "brighterMonday": Object {
                "title": "Brighter Monday (East Africa)",
                "url": "https://www.brightermonday.com/",
              },
              "catho": Object {
                "title": "Catho (Brazil)",
                "url": "https://www.catho.com.br/",
              },
              "title": "International partners",
              "zhaopin": Object {
                "title": "Zhaopin (China)",
                "url": "https://www.zhaopin.com/",
              },
            },
            "privacy": Object {
              "title": "Privacy Statement",
              "url": "/pages/terms/privacy-policy",
            },
            "rightReserved": "All Rights Reserved.",
            "siteMap": Object {
              "title": "Site Map",
              "url": "/sitemap",
            },
            "terms": Object {
              "title": "Terms & Conditions",
              "url": "/pages/terms/terms-conditions",
            },
          }
        }
        linkRenderer={[Function]}
      />
      <Hidden
        aboveMobile={true}
        className=""
        component="span"
        desktop={false}
        mobile={false}
        print={false}
        screen={false}
      >
        <FooterLinks
          footerMessages={
            Object {
              "about": Object {
                "about": Object {
                  "title": "About jobsDB",
                  "url": "/pages/aboutus",
                },
                "contactUs": Object {
                  "mobileUrl": "https://m.jobsdb.com/en-hk/legal.do?p=contact_us",
                  "title": "Contact us",
                  "url": "/pages/regional-coverage",
                },
                "title": "About JobsDB",
              },
              "cfsLink": "/hk/en",
              "cmsLink": "/en-hk",
              "connect": Object {
                "appStore": Object {
                  "icon": "apple",
                  "title": "jobsDB @ App Store",
                  "url": "https://itunes.apple.com/hk/app/jobsdb-job-search/id414607432",
                },
                "playStore": Object {
                  "icon": "android",
                  "title": "jobsDB @ Google Play",
                  "url": "https://play.google.com/store/apps/details?id=com.jobsdb",
                },
                "title": "Connect",
              },
              "copyright": "Copyright © 1998-{year}, jobsDB.",
              "legal": Object {
                "title": "Legal terms",
                "url": "/legal.do",
              },
              "partners": Object {
                "bdjobs": Object {
                  "title": "Bdjobs (Bandladesh)",
                  "url": "https://www.bdjobs.com/",
                },
                "brighterMonday": Object {
                  "title": "Brighter Monday (East Africa)",
                  "url": "https://www.brightermonday.com/",
                },
                "catho": Object {
                  "title": "Catho (Brazil)",
                  "url": "https://www.catho.com.br/",
                },
                "title": "International partners",
                "zhaopin": Object {
                  "title": "Zhaopin (China)",
                  "url": "https://www.zhaopin.com/",
                },
              },
              "privacy": Object {
                "title": "Privacy Statement",
                "url": "/pages/terms/privacy-policy",
              },
              "rightReserved": "All Rights Reserved.",
              "siteMap": Object {
                "title": "Site Map",
                "url": "/sitemap",
              },
              "terms": Object {
                "title": "Terms & Conditions",
                "url": "/pages/terms/terms-conditions",
              },
            }
          }
          linkRenderer={[Function]}
        />
      </Hidden>
    </div>
    <div
      className="lowerWrapper"
    >
      <div
        className="meta"
      >
        <Hidden
          aboveMobile={false}
          className=""
          component="span"
          desktop={false}
          mobile={true}
          print={false}
          screen={false}
        >
          <FooterLinks
            className="bottomMeta"
            footerMessages={
              Object {
                "about": Object {
                  "about": Object {
                    "title": "About jobsDB",
                    "url": "/pages/aboutus",
                  },
                  "contactUs": Object {
                    "mobileUrl": "https://m.jobsdb.com/en-hk/legal.do?p=contact_us",
                    "title": "Contact us",
                    "url": "/pages/regional-coverage",
                  },
                  "title": "About JobsDB",
                },
                "cfsLink": "/hk/en",
                "cmsLink": "/en-hk",
                "connect": Object {
                  "appStore": Object {
                    "icon": "apple",
                    "title": "jobsDB @ App Store",
                    "url": "https://itunes.apple.com/hk/app/jobsdb-job-search/id414607432",
                  },
                  "playStore": Object {
                    "icon": "android",
                    "title": "jobsDB @ Google Play",
                    "url": "https://play.google.com/store/apps/details?id=com.jobsdb",
                  },
                  "title": "Connect",
                },
                "copyright": "Copyright © 1998-{year}, jobsDB.",
                "legal": Object {
                  "title": "Legal terms",
                  "url": "/legal.do",
                },
                "partners": Object {
                  "bdjobs": Object {
                    "title": "Bdjobs (Bandladesh)",
                    "url": "https://www.bdjobs.com/",
                  },
                  "brighterMonday": Object {
                    "title": "Brighter Monday (East Africa)",
                    "url": "https://www.brightermonday.com/",
                  },
                  "catho": Object {
                    "title": "Catho (Brazil)",
                    "url": "https://www.catho.com.br/",
                  },
                  "title": "International partners",
                  "zhaopin": Object {
                    "title": "Zhaopin (China)",
                    "url": "https://www.zhaopin.com/",
                  },
                },
                "privacy": Object {
                  "title": "Privacy Statement",
                  "url": "/pages/terms/privacy-policy",
                },
                "rightReserved": "All Rights Reserved.",
                "siteMap": Object {
                  "title": "Site Map",
                  "url": "/sitemap",
                },
                "terms": Object {
                  "title": "Terms & Conditions",
                  "url": "/pages/terms/terms-conditions",
                },
              }
            }
            linkRenderer={[Function]}
          />
        </Hidden>
        <Text
          baseline={true}
          className="copyright"
          secondary={true}
          semiStrong={true}
          whispering={true}
        >
          Copyright © 1998-2020, jobsDB.
        </Text>
      </div>
      <div
        className="chevronIcon"
        onClick={[Function]}
      >
        <Icon
          animation=""
          id="pageBottom"
          lineHeight=""
          rotation="-180deg"
          size=""
          smoothRotate={true}
          type="chevron"
        />
      </div>
    </div>
  </PageBlock>
</footer>
`;
